sword2/utils.py

```diff
@@ -55,17 +51,18 @@ def get_md5(data):
     """
     if hasattr(data, "read") and hasattr(data, 'seek'):
         m = md5()
-        chunk = data.read(1024*1024)   # 1Mb
         f_size = 0
-        while(chunk):
-            f_size += len(chunk)
+        for chunk in iter(lambda: data.read(1024*1024), b""):
```

Comment:
this seems very obscure to me. What's going on here?

Reply:
it reads a file as bytecodes and generates hash code in chunks. Iterating the chunks in a regular while loop causes a bytestring incompatibility somewhere else in the code. So, I changed it to this `iter(lambda: data.read(1024*1024), b"")`.